asciidoc/courses/llm-fundamentals/modules/1-introduction/lessons/2-hallucination/lesson.adoc

```diff
@@ -0,0 +1,129 @@
+On the other hand, a lower temperature, even `0`, is required when a response grounded in facts is important.
+
+[TIP]
+.Consider the correct temperature
+====
+In June 2023, two US lawyers were link:https://www.reuters.com/legal/new-york-lawyers-sanctioned-using-fake-chatgpt-cases-legal-brief-2023-06-22/[sanctioned for submitting an LLM-generated legal brief] that contained six fictitious case citations.
+====
+
+A quick fix _may_ be to reduce the temperature.  But more likely, the LLM is hallucinating because it hasn't got the information required.
+
+
+=== Model Training and Complexity
+
+Large Language Models (LLMs) are often considered "black boxes" due to the difficulty in deciphering their decision-making processes. The complexity of these models combined with potential training on erroneous or misleading data means that their outputs can sometimes be unpredictable or inaccurate.
+
+For example, if asked about a controversial historical event, an LLM might produce an answer that's biased or misrepresentative, largely due to the data it was trained on. Furthermore, it would be near impossible to transparently trace back how the model arrived at that conclusion.
+
+
+
+=== Missing Information
+
+The training process for LLMs is intricate and time-intensive, often requiring vast datasets compiled over extended periods. As such, these models might lack the most recent information or might miss out on specific niche topics not well-represented in their training data.
+
+For instance, if an LLM's last update was in September 2022, it would be unaware of world events or advancements in various fields that occurred post that date, leading to potential gaps in its knowledge or responses that seem out of touch with current realities.
+
+If the user is asking a question on information that is hard to find, or outside of the public domain, it will be next to impossible for an LLM to provide an accurate response.
+
+Luckily, this is where a knowledge graph comes in.
+
+
+// How?
+== Improving LLM Accuracy
+
+The following methods can be employed to help guide LLMs to produce more consistent and accurate results.
+
+
+=== Prompt Engineering
+
+Prompt engineering is an iterative process of crafting specific and deliberate questions that guide the LLM toward the desired response.
+By refining the way questions are posed, developers can achieve better results from existing models without the need for retraining.
+
+For example, if you require a summary of a blog post, rather than asking _"What is this book about?"_, a more appropriate response would be _"Provide a concise, three-sentence summary and three tags for this blog post."_
+
+You could even throw in _"Return the response as JSON"_ and provide an example output to make it easier to parse in the programming language of your choice.
+
+[TIP]
+.Be Positive
+====
+When writing a prompt, I aim to provide positive instructions.
```

Comment:
or tell the model to take a breadth
or tell it when it doesn't follow instructions bad things will happen